### DISCUSSION TITLE
3.4.1 - Port 443 permission denied

### DISCUSSION THREAD

**Discussion by ghost** [coder/code-server#1774] [Q&A] [answered]
Since version 3.4.1 I haven’t been able to run code-server without root permissions. Also, when using root my configuration file doesn’t seem to be used. For code-server to work I have to manually set my configuration as flags.
I’m on the Raspberry Pi 4 with Raspbian 64 bit.

```
pi@raspberrypi:~ $ cat ~/.config/code-server/config.yaml
bind-addr: 0.0.0.0:443
auth: none
password: ***
cert: "/home/pi/raspberrypi.local.pem"
cert-key: "/home/pi/raspberrypi.local-key.pem"

pi@raspberrypi:~ $ code-server 
info  Using config file ~/.config/code-server/config.yaml
info  Using user-data-dir ~/.local/share/code-server
info  code-server 3.4.1 48f7c2724827e526eeaa6c2c151c520f48a61259
error listen EACCES: permission denied 0.0.0.0:443

pi@raspberrypi:~ $ sudo code-server 
info  Using config file ~/.config/code-server/config.yaml
info  Using user-data-dir ~/.local/share/code-server
info  code-server 3.4.1 48f7c2724827e526eeaa6c2c151c520f48a61259
info  HTTP server listening on http://127.0.0.1:8080
info      - Using password from ~/.config/code-server/config.yaml
info      - To disable use `--auth none`
info    - Not serving HTTPS
^C
pi@raspberrypi:~ sudo lsof -i -P -n | grep LISTEN
sshd       520  root    3u  IPv4  15148      0t0  TCP *:22 (LISTEN)
sshd       520  root    4u  IPv6  15150      0t0  TCP *:22 (LISTEN)

pi@raspberrypi:~ $ sudo code-server --bind-addr 0.0.0.0:443 --auth none --cert "/home/pi/raspberrypi.local.pem" --cert-key "/home/pi/raspberrypi.local-key.pem"
info  Using config file ~/.config/code-server/config.yaml
info  Using user-data-dir ~/.local/share/code-server
info  code-server 3.4.1 48f7c2724827e526eeaa6c2c151c520f48a61259
info  HTTP server listening on https://0.0.0.0:443
info    - No authentication
info    - Using provided certificate and key for HTTPS
```

EDIT: Readability.

**Comment by nhooyr** [comment, marked as answer]
@dinisbcoelho So to run on `:443` root permissions are required. In your case, the problem is that sudo modifies `$HOME` to the root's home and your config file isn't being used.

```
sudo env HOME=$HOME code-server
```

will work as expected.

**Comment by nhooyr** [comment]
See https://unix.stackexchange.com/questions/91384/how-is-sudo-set-to-not-change-home-in-ubuntu-and-how-to-disable-this-behavior for more details

**Comment by sprucecellodev125** [comment]
well, this might be the only one solution, but also insecure when I wanted to give someone access to code-server in separated user. I want to give them code server by creating non-root user to prevent doing system-wide change and accessing my home directory. I set the port to 8081 but still the same error. Can someone recommend me secure solution without giving them root access?
